src/gtest/test_joinsplit.cpp

```diff
@@ -0,0 +1,213 @@
+#include <gtest/gtest.h>
+
+#include "utilstrencodings.h"
+
+#include "zcash/prf.h"
+
+#include "zcash/JoinSplit.hpp"
+#include "zcash/Note.hpp"
+#include "zcash/NoteEncryption.hpp"
+#include "zcash/IncrementalMerkleTree.hpp"
+
+using namespace libzcash;
+
+void test_full_api(ZCJoinSplit* js)
+{
+    // The recipient's information.
+    SpendingKey recipient_key = SpendingKey::random();
+    PaymentAddress recipient_addr = recipient_key.address();
+
+    // Create the commitment tree
+    ZCIncrementalMerkleTree tree;
+
+    // Set up a JoinSplit description
+    uint256 ephemeralKey;
+    uint256 randomSeed;
+    uint64_t vpub_old = 10;
+    uint64_t vpub_new = 0;
+    uint256 pubKeyHash = random_uint256();
+    boost::array<uint256, 2> hmacs;
```

Comment:
Rename `hmacs` to something else since we aren't using HMAC (do this everywhere).